### DISCUSSION TITLE
Is API head filter even working?

### DISCUSSION THREAD

**Discussion by tsipa** [community/community#24354] [Apps, API and Webhooks]
<p>Hello, i’m trying to fetch PRs by head attribute.</p>
<p>I’m doing :<br>
<code>curl 'https://api.github.com/repositories/284089292/pulls?state=closed&amp;base=bpf-next&amp;head=series/357901=&gt;bpf-next' | grep head -A 2</code><br>
I’m expecting:<br>
the result will contain only closed pulls sent from head “series/357901=&gt;bpf-next” to base “bpf-next”<br>
like in UI:<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/tsipa/bpf-next/pulls?q=is:pr%2Bstate:closed%2Bbase:bpf-next%2Bhead:series%2F357901=%3Ebpf-next" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/3116070/181084332-58cd9ef1-428b-4360-b4d2-97cf25f8c4ba.png" class="thumbnail onebox-avatar" width="420" height="420">

<h3><a href="https://github.com/tsipa/bpf-next/pulls?q=is:pr%2Bstate:closed%2Bbase:bpf-next%2Bhead:series%2F357901=%3Ebpf-next" target="_blank" rel="noopener nofollow ugc">tsipa/bpf-next</a></h3>

<p>Contribute to tsipa/bpf-next development by creating an account on GitHub.</p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>What i’ve got:<br>
i’ve got all closed PR for base “bpf-next”.</p>
<p>inb4 - i’ve tried to urlencode, the <code>https://api.github.com/repositories/284089292/pulls?state=closed&amp;base=bpf-next&amp;head=series%2F357901%3D%3Ebpf-next</code> it makes no difference.</p>

**Comment by nethgato** [comment, marked as answer]
<p>Hey <a class="mention" href="/tsipa">@tsipa</a> :wave:</p>
<p>For the <code>head-ref</code>  you need to include either the <code>user</code> or <code>org</code> value like: <code>tsipa:series/357901=&gt;bpf-next</code>. We call it out in our docs, here:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls#list-pull-requests" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="" height="">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls#list-pull-requests" target="_blank" rel="noopener">Pulls - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>…where:</p>
<blockquote>
<p>Filter pulls by head user or head organization and branch name in the format of  <code>user:ref-name</code>  or  <code>organization:ref-name</code> . For example:  <code>github:new-script-format</code>  or  <code>octocat:test-branch</code> .</p>
</blockquote>
<p>Give this a try:</p>
<pre><code class="lang-auto">curl -L -X GET 'https://api.github.com/repos/tsipa/bpf-next/pulls?state=closed&amp;base=bpf-next&amp;head=tsipa:series/357901=%3Ebpf-next'
</code></pre>
<p>Should work! :crossed_fingers:</p>
